src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/WPSExecutionManager.java

```diff
@@ -450,12 +478,15 @@ private ExecuteResponseType execute() {
                             inputPercentage + executionPercentage,
                             "Execution completed, preparing to write response");
                 }
+                chainNotifier.fireReturned();
             } catch (ProcessDismissedException e) {
                 // that's fine, move on writing the output
+                chainNotifier.fireFailed();
```

Comment:
This is not a failure though, it's the user asking to dismiss an otherwise properly running process.

Reply:
It's not going to run further chained processes tho, which is what the listener is about. Maybe different method names are needed here.